### ISSUE TITLE
A way to set Mozilla preferences

### ISSUE THREAD

**Issue by MarkR42** [laurentj/slimerjs#385]
As a potential enhancement, it would be really useful to set Mozilla preferences programmatically.

**Comment by marco-c** [comment]
There is a way to do this, by using the `chrome` module:
https://github.com/mozilla/pluotsorbet/blob/9afd89c0079833245de4447fd9dd13366bbc69dd/tests/automation.js#L8

**Comment by MarkR42** [comment]
Yes, I see that, I tried using chrome module to set settings, it does work.

I suppose we should update the documentation to give this example.

Mark

**Comment by danikp** [comment]
@MarkR42 can you post an example code of setting preferences using chrome module ?

**Comment by MarkR42** [comment]
Dmitry, Hi,

This works for me, but there might be a nicer way:

    var {Cc, Ci} = require("chrome");
     var prefs = Cc["@mozilla.org/preferences-service;1"]
                    .getService(Ci.nsIPrefService);
    var branch = prefs.getBranch("privacy.donottrackheader.");
    branch.setBoolPref("enabled", true);

This example sets the "Do not track" flag, which can be tested easily
because web servers receive it as a HTTP header "DNT".

Unfortunately very little of the add-on SDK is shipped with SlimerJS, so
it's not available (Firefox's own versions, cannot be require()d).

I originally tried: var prefs_service=require("sdk/preferences/service");
prefs_service.set(...

That does NOT work in SlimerJS.

However, because the Firefox addon-SDK is mostly wrappers for XPCOM stuff
(as in the example above), it can be used directly.

Mark
